app/routers/accounts.py

```diff
@@ -116,16 +159,58 @@ async def get_accounts(account_repo: AccountRepository):
     return await paginate(account_repo, AccountRead)
 
 
-@router.put("/{id}", response_model=AccountRead)
-async def update_account(id: str, data: AccountUpdate):
-    pass
+@router.put(
+    "/{id}",
+    response_model=AccountRead,
+    dependencies=[Depends(check_operations_account)],
+)
+async def update_account(
+    data: AccountUpdate,
+    account_repo: AccountRepository,
+    account: Annotated[Account, Depends(fetch_account_or_404)],
+):
+    """
+    This Endpoint updates an Affiliate Account.
+
+    The following conditions must be verified before proceeding with the operation of updating
+    an account.
+    1. The Account type must be OPERATIONS, otherwise a 403 error will be returned
+    2. The Account status must be Active
+    3. Only Accounts classified as of type “Affiliate” can be updated.
+    4. Only the name and the external_id of the account can be modified.
+
+    The newly created account's status will be assigned as ACTIVE
```

Comment:
1. This operation can be performed only in the context of an Operations Account otherwise 403 will be returned.
2. Deleted Accounts cannot be updated.

The sentence

"The newly created account's status will be assigned as ACTIVE" looks copy/pasted from the create endpoint.